docs/specs/component.md

```diff
@@ -105,99 +108,121 @@ events:
 
 #### ComponentBytesReceived
 
-*Sources* MUST emit a `ComponentBytesReceived` event immediately after receiving, decompressing
-and filtering bytes from the upstream source and before the creation of a Vector event.
+*Sources* MUST emit a `ComponentBytesReceived` event that represents the
+reception of *raw network bytes*. This event is intended to reflect the raw
+network usage for the source.
 
+- Emission
+  - MUST emit immediately after reception of raw network bytes.
+  - MUST emit *before* mutation of the bytes, such as decryption, decompression,
+    and filtering.
 - Properties
-  - `byte_size`
-    - For UDP, TCP, and Unix protocols, the total number of bytes received from
-      the socket excluding the delimiter.
-    - For HTTP-based protocols, the total number of bytes in the HTTP body, as
-      represented by the `Content-Length` header.
-    - For files, the total number of bytes read from the file excluding the
-      delimiter.
-  - `protocol` - The protocol used to send the bytes (i.e., `tcp`, `udp`,
-    `unix`, `http`, `https`, `file`, etc.)
-  - `http_path` - If relevant, the HTTP path, excluding query strings.
-  - `socket` - If relevant, the socket number that bytes were received from.
+  - `byte_size` - REQUIRED, number of raw bytes received.
+    - SHOULD be the total number of raw network bytes received *including*
+      framing (headers, delimiters, etc.). If If this is not possible due to a
+      client limitation, then the framed bytes MUST be used and the `framed` tag
+      MUST be set to `true`. 
```

Comment:
So I find this wording a little confusing.

Based on the definition below of `framed`, I would expect `byte_size` to include the size of the entire request -- framing, headers, payload, all of it -- when `framed` is `true`... but here, you're saying that it's not possible to include the bytes related to framing/headers, then the "framed bytes" (huh? does this mean payload only?) must be used and `framed` must be set to `true`.

Overall, I think there needs to be a change to the wording to talk about the bytes as raw vs processed, _then_ refer to "framing" as all the additional bytes on top of the bytes we have to process aka decode, and then clear up the weird conditional statement about what `framed` means.

Reply:
I agree. See https://github.com/vectordotdev/vector/pull/12912#issuecomment-1206572535 for more info.